app/tests/permissions/delete-client.test.js

```diff
@@ -0,0 +1,57 @@
+const test = require('ava')
+const db = require('../../db')
+const path = require('path')
+const hydra = require('../../lib/hydra')
+const sinon = require('sinon')
+
+const migrationsDirectory = path.join(__dirname, '..', '..', 'db', 'migrations')
+
+let agent
+test.before(async () => {
+  const conn = await db()
+  await conn.migrate.latest({ directory: migrationsDirectory })
+  await conn.schema.createTable('hydra_client', t => {
```

Comment:
there should be a migration for this and creating/deleting clients in the db should be managed as part of lib/client https://github.com/developmentseed/osm-teams/blob/master/app/lib/hydra.js#L100-L138

alternately maybe it's easiest to make a request to the hydra api to check the delete permission.

Reply:
We already have a migration for this as part of the `hydra migrations`. We just don't manage them because they are part of the hydra set up. We _could_ set up the testing framework to import all the hydra tables, but it seems like a lot.

I disagree that clients should have their own table on the app side, because then for each insert into hydra, we would have to also insert into our own table. 

We can make a hydra getClient request to see who the owner of the app is, but if we have access to the database, why not use it @sethvincent ?

Reply:
oh got it, this is a table created/used by hydra. never mind!